### PR TITLE
ICLD-1615: Implement OAuth2 redirect login for Swagger UI

### PR DESCRIPTION
**Pull request by lvbnhj**

### CODE REVIEW

**Comment by lvbnhj** [comment]
@donvadicastro, please review

File: src/main/javascript/router.js [hunk 1]
```diff
@@ -38,7 +46,21 @@ window.SwaggerUiRouter = Backbone.Router.extend({
...
-                    Backbone.history.navigate('login', true);
+                    var host = window.location;
+                    var pathname = location.pathname.substring(0, location.pathname.lastIndexOf('/'));
+                    var url = host.protocol + '//' + host.host + pathname.replace('swagger', 'login/url');
+                    $.ajax({
+                        url : url,
+                        type: 'POST',
+                        success: function (data)
+                        {
+                            window.location.href = data;
```
Comment (donvadicastro) [inline src/main/javascript/router.js:57]: We are using here full page reload? Or we can use Backbone.history for partial updates only?

File: src/main/javascript/router.js [hunk 2]
```diff
@@ -134,7 +187,7 @@ window.SwaggerUiRouter = Backbone.Router.extend({
             console.log('render documentation page');
             this.showView(new SwaggerUi.Views.DocumentationView(subdoc && { template: 'documentation_' + subdoc }));
         } else {
-            this.navigate('login', true);
+            this.navigate('', true);
```
Comment (donvadicastro) [inline src/main/javascript/router.js:190]: Is logout process works as expected just navigating to default page?

File: src/main/javascript/router.js [hunk 3]
```diff
@@ -154,5 +207,23 @@ window.SwaggerUiRouter = Backbone.Router.extend({
         var pathname = location.pathname.substring(0, location.pathname.lastIndexOf('/'));
 
         return host.protocol + '//' + host.host + pathname.replace('swagger', 'api/swagger/docs/v1') + '?_=' + Date.now();
+    },
+
+    getParameterByName: function(name, url) {
+        if (!url) {
+            url = window.location.href;
+        }
+        name = name.replace(/[\[\]]/g, '\\$&');
```
Comment (donvadicastro) [inline src/main/javascript/router.js:216]: Could you please add some comments to understand in future how to extend this function? Details about what regex describes in this example.

**Review by donvadicastro — APPROVED** [review]

File: src/main/javascript/router.js (hunk 1 above)
Comment (lvbnhj) [reply src/main/javascript/router.js:57]: Yep, here we have redirected to the authorization site

File: src/main/javascript/router.js (hunk 2 above)
Comment (lvbnhj) [reply src/main/javascript/router.js:190]: Yes, and then navigates to the authorization server again

File: src/main/javascript/router.js (hunk 3 above)
Comment (lvbnhj) [reply src/main/javascript/router.js:216]: it splits query string by #, ? and & and then extracts corresponding query string parameter value

**Pull request merged**